main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -0,0 +1,354 @@
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.CompilationMessage
+import ca.uwaterloo.flix.language.ast.LiftedAst.Expression
+import ca.uwaterloo.flix.language.ast.{LiftedAst, OccurrenceAst, Symbol}
+import ca.uwaterloo.flix.util.Validation
+import ca.uwaterloo.flix.util.Validation.ToSuccess
+import ca.uwaterloo.flix.language.ast.OccurrenceAst.Occur._
+import ca.uwaterloo.flix.language.ast.OccurrenceAst.Occur
+
+object OccurrenceAnalyzer {
+
+
+  def run(root: LiftedAst.Root)(implicit flix: Flix): Validation[OccurrenceAst.Root, CompilationMessage] = flix.phase("OccurrenceAnalyzer") {
+    // TODO: Implement occurrence analysis.
+    val defs = root.defs.map {
+      case (sym, defn) =>
+        val fparams = defn.fparams.map { case LiftedAst.FormalParam(sym, mod, tpe, loc) => OccurrenceAst.FormalParam(sym, mod, tpe, loc) }
+        val (e, _) = visitExp(defn.exp)
+        sym -> OccurrenceAst.Def(defn.ann, defn.mod, defn.sym, fparams, e, defn.tpe, defn.loc)
+    }
+    val enums = root.enums.map {
```

Comment:
What is this computation for?

Reply:
Added comment explaining that it transforms LiftedAst Defs to OccurrenceAst Defs